2016/exercises/final_project/Franka-Tim/poging2.py

```diff
@@ -0,0 +1,243 @@
+            for k in [j+1..len(galeDiagram)-1]:
+                #print(galeDiagram[i],galeDiagram[j],galeDiagram[k]) #Check that the functions indeed checks all combinations
+                print(galeDiagram[i],galeDiagram[j],galeDiagram[k])
+                if checkCollinear(galeDiagram[i],galeDiagram[j],galeDiagram[k]) == True:
+                    isSimplicial = False
+                    #We want to break the loop here but Python doesn't allow us to do so :'(
+                    #This means that we have to check all 8 choose 3 = 56 combinations
+    return isSimplicial
+
+def checkCollinear(i,j,k):  #Check for every combination of 3 points whether or not they are collinear
+    a = i[0]
+    b = i[1]
+    c = j[0]
+    d = j[1]
+    e = k[0]
+    f = k[1]
+    M = matrix(3, 3, [1, a, b , 1,c,d, 1,e,f])
+    #print(M)
+    if numpy.linalg.det(M) == 0:
+        return True
+    else: return False
+
+
+##### testlijsten #####
+lijst1 = [[1,1,'-'],[2,2,'+'],[3,4,'+'],[4,5,'+'],[5,6,'-'],[4,0,'+'],[6,1,'-'],[3,5,'+']]
+
+lijst2 = [[1,1,'-'],[2,2,'+'],[3,4,'+'],[4,5,'+']]
+
+##### Generate random Gale Diagrams #####
+#Generate all combinations of + and - of length 8
+signs = ['+','-']
+allCombinations = [p for p in itertools.product(signs, repeat=8)]
+
+# Count the number of '+' signs in a given combination
+# A bit unnecessary because we already had a function for this made for input length 4..
+
+def combCounter(combo):
+    counter = 0
+    if len(combo) != 8:
+        print("Input length does not equal 8.")
+    else:
+        for i in [0..len(combo)-1]:
+            dummy = combo[i]
+            if dummy == '+':
+                counter += 1
+    return counter
+
+#Filter out all indices of combinations that have 3 or 4 '+' signs, because we know a Gale Diagram corresponding to a neighborly polytope has 3 or 4 '+' (or '-') signes
+
+def checkCombinations():
+    combList = list()
+    for i in [0..255]:
+        cur = allCombinations[i]
+        if (combCounter(cur) == 3) or (combCounter(cur) == 4):
+            combList.append(i)
+    return combList
+
+
+goodCombs = checkCombinations()
+len(goodCombs)
+
+#Generate the coordinates at random and append a correct signlist (index as input) to generate a Gale Diagram
+
+def generateRandomGale(integer):
+    randomGD = list()
+    goodCombs = checkCombinations()
+    randomGD.append([random.randrange(-5, 5) for _ in range(0, 8)])
+    randomGD.append([random.randrange(-5, 5) for _ in range(0, 8)])
+    k = goodCombs[integer]
+    randomGD.append(list(allCombinations[k]))
+    return randomGD
+
+#Transpose matrix
+uitkomst = generateRandomGale(4) #Example to check if it works
+randomGDt = map(list,map(None,*uitkomst))
+
+#generate a random coordinate and a random correct signlist to generate a random gale diagram
+
+def mainFunction():
+    neighbourlyGD = list()
+    for i in [0..5]: #Generate 500 random points
+        rand = random.randint(0,125) #Pick a random sign configuration

```

Comment:
We've put all indices corresponding to 'good' (i.e. #+ in the configuration = 3 or 4) sign configurations in a list, which had length 126